src/main/java/uk/gov/hmcts/reform/rse/idam/simulator/service/SimulatorService.java

```diff
@@ -62,30 +62,34 @@ public String generateACachedToken(String userName, String clientID, String gran
                 "Idam Simulator: No User for this userName " + userName
             );
         }
-        Boolean tokenExpired = liveMemoryService.getByEmail(userName).stream()
+        Boolean tokenExpired = userService.getByEmail(userName).stream()
             .map(SimObject::getMostRecentJwTokenUnixTime)
             .map(t -> System.currentTimeMillis() > t + tokenExpirationMs).findFirst().get();
 
-        if (tokenExpired || liveMemoryService.getByEmail(userName).get().getMostRecentJwToken() == null) {
+        if (tokenExpired || userService.getByEmail(userName).get().getMostRecentJwToken() == null) {
             LOG.info("Token not existing or expired and will be regenerated");
             return jwTokenGenerator.generateToken(issuer, tokenExpirationMs, userName, clientID, grantType);
         }
         LOG.info("Use token in cache");
-        return liveMemoryService.getByEmail(userName).get().getMostRecentJwToken();
+        return userService.getByEmail(userName).get().getMostRecentJwToken();
     }
 
     public void updateTokenInUser(String username, String token) {
         Optional<SimObject> userInMemory = checkUserInMemoryNotEmptyByUserName(username);
-        userInMemory.get().setMostRecentJwToken(token);
+        SimObject user = userInMemory.get();
+        user.setMostRecentJwToken(token);
+        userService.putSimObject(user.getId(), user);
```

Comment:
To ensure that the updated objects always get committed when using persistent storage too.